### PR TITLE
[pagination] followups to clean up search registered models

### LINKED ISSUE #2976: [BUG] `mlflow.search_runs` does not raise error or warning when passed invalid `order_by` expression
`mlflow.search_runs` does not raise error or warning when passed invalid `order_by` expression. I think this behavior is dangerous. Let's consider a scenario where a user wants to select the best model from a bunch of runs and writes the code below:

```python
order_by = ["metrics.accuracy DECS"]  # "DECS" is a typo, should be "DESC"
sorted_runs = mlflow.search_runs(..., order_by=order_by)
best_run = sorted_runs.iloc[0]  # select the frist row
best_model = load_model_from_run(best_run)
```

What this code does is:
- The `order_by` expression `"metrics.accuracy DECS"` doesn't end with ` desc` or ` asc` so it's parsed as a metric named `accuracy DECS`.
- `search_runs` sorts runs by the metric `accuracy DECS` which doesn't exist and returns unsorted runs.
- `sorted_runs.iloc[0]` is obviously not the best run, **but no error or warning is raised**.

### PR DESCRIPTION
**Pull request by ankit-db**
## What changes are proposed in this pull request?

This PR cleans up some issues with testing and edge cases with search registered models:

1. It adds tests to verify that lowercase and uppercase ordering keys work.
2. It changes search `order_by` parsing in `search_runs` and `search_registered_models` to throw if an invalid token is passed in

per discussion with @mparkhe and @sueann 

## How is this patch tested?

Unit tests and manual

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/search_utils.py [hunk 1]
```diff
@@ -363,11 +366,20 @@ def _validate_order_by_and_generate_token(cls, order_by):
...
+        elif len(tokens) == 2:
+            order_token = tokens[1].lower()
+            if order_token not in cls.VALID_ORDER_BY_TAGS:
+                raise MlflowException(f"Invalid ordering key in order_by clause '{order_by}'.",
+                                      error_code=INVALID_PARAMETER_VALUE)
+            if order_token == cls.DESC_OPERATOR:
+                is_ascending = False
+                token_value = tokens[0]
+            elif order_token == cls.ASC_OPERATOR:
+                token_value = tokens[0]
```
Comment (sueann) [inline mlflow/utils/search_utils.py:382]: nit: could do
```
is_ascending  = (order_token == cls.ASC_OPERATOR)
token_value = tokens[0]
```

File: tests/store/model_registry/test_sqlalchemy_store.py [hunk 2]
```diff
@@ -515,6 +515,11 @@ def test_search_registered_models(self):
         rms, _ = self._search_registered_models(f"name iLike '%blah%'")
         self.assertEqual(rms, [])
 
+        # confirm that ILIKE works for empty query
+        rms, _ = self._search_registered_models(f"name iLike '%%'")
+        # raise Exception()
```
Comment (sueann) [inline tests/store/model_registry/test_sqlalchemy_store.py:520]: ?

File: mlflow/utils/search_utils.py [hunk 3]
```diff
@@ -363,11 +366,20 @@ def _validate_order_by_and_generate_token(cls, order_by):
...
-        elif token_value.lower().endswith(" asc"):
-            token_value = token_value[0:-len(" asc")]
+        tokens = token_value.split()
+        if len(tokens) > 2:
+            raise MlflowException(f"Invalid order_by clause '{order_by}'. Could not be parsed.",
+                                  error_code=INVALID_PARAMETER_VALUE)
+        elif len(tokens) == 2:
+            order_token = tokens[1].lower()
+            if order_token not in cls.VALID_ORDER_BY_TAGS:
+                raise MlflowException(f"Invalid ordering key in order_by clause '{order_by}'.",
```
Comment (sueann) [inline mlflow/utils/search_utils.py:376]: this does cause a behavior change for `search_runs` (so this is technically a breaking change), right? i see that lines 355-362 above check the ordering keywords in some way, so just wanted to confirm.

**Review by sueann — COMMENTED** [review]
just a few nits + 1 question, otherwise lgtm - thanks!

**Comment by sueann** [comment]
also, i looked at the unit tests a bit more and we should do a couple of things:
* add the new error cases to https://github.com/mlflow/mlflow/blob/230c7d722c9d52522a1b7b4ce76b79b86273c190/tests/utils/test_search_utils.py#L323-L326
* add a similar test case for `parse_order_by_for_search_registered_models` in the same test suite^

File: mlflow/utils/search_utils.py (hunk 3 above)
Comment (ankitmathur-db) [reply mlflow/utils/search_utils.py:376]: Yeah that's true - it is breaking

File: tests/store/model_registry/test_sqlalchemy_store.py (hunk 2 above)
Comment (ankitmathur-db) [reply tests/store/model_registry/test_sqlalchemy_store.py:520]: This is the query that the UI sends when the page first loads - the comment raising Exception is something I missed while debugging. Removed now

**Comment by ankitmathur-db** [comment]
Thanks for the review @sueann - updated the PR based on your comments

**Review by sueann — APPROVED** [review]
lgtm!

**Pull request merged**